torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,454 @@
+        )
+
+        # Filter aux_df according to min_target_value
+        if self.min_feature_value is not None:
+            for feature, min_value in self.min_feature_value.items():
+                self.aux_df = self.aux_df[self.aux_df[feature] >= min_value]
+
+        # Filter aux_df according to max_target_value
+        if self.max_feature_value is not None:
+            for feature, max_value in self.max_feature_value.items():
+                self.aux_df = self.aux_df[self.aux_df[feature] <= max_value]
+
+        # collect target mean and std for each target
+        self.target_mean: dict[str, float] = self.aux_df[self.targets].mean().to_dict()
+        self.target_std: dict[str, float] = self.aux_df[self.targets].std().to_dict()
+
+        def _get_subsequences(df: pd.DataFrame, k: int) -> list[dict[str, list[int]]]:
+            """Generate all possible subsequences of length k for a given group.
+
+            Args:
+                df: grouped dataframe of a single typhoon
+                k: length of the subsequences to generate
+
+            Returns:
+                list of all possible subsequences of length k for a given typhoon id
+            """
+            min_seq_id = df['seq_id'].min()
+            max_seq_id = df['seq_id'].max()
+
+            # generate possible subsquences of length k for group
+            subsequences = [
+                {'id': df['id'].iloc[0], 'seq_id': list(range(i, i + k))}
+                for i in range(min_seq_id, max_seq_id - k + 2)
+            ]
+            return [
+                subseq
+                for subseq in subsequences
+                if set(subseq['seq_id']).issubset(df['seq_id'])
+            ]
+
+        self.sample_sequences: list[_SampleSequenceDict] = [
+            item
+            for sublist in self.aux_df.groupby('id')
+            .apply(_get_subsequences, k=self.sequence_length)
+            .tolist()
+            for item in sublist
+        ]
+
+    def __getitem__(self, index: int) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data, labels, and metadata at that index
+        """
+        sample_entry = self.sample_sequences[index]
+        sample_df = self.aux_df[
+            (self.aux_df['id'] == sample_entry['id'])
+            & (self.aux_df['seq_id'].isin(sample_entry['seq_id']))
+        ]
+
+        sample = {'image': self._load_image(sample_df)}
+        # load features of the last image in the sequence
+        sample.update(
+            self._load_features(
+                os.path.join(
+                    self.root,
+                    self.data_root,
+                    'metadata',
+                    str(sample_df.iloc[-1]['id']) + '.csv',
+                ),
+                sample_df.iloc[-1]['image_path'],
+            )
+        )
+
+        # torchgeo expects a single label
```

Comment:
We can fix this if needed. I'm trying to add support for multi-label classification in #2219